includes/class-sensei-lesson.php

```diff
@@ -3325,52 +3326,58 @@ public function lesson_quiz_questions( $quiz_id = 0, $post_status = 'any', $orde
 					$questions[] = $question;
 				}
 			} else {
-				// Otherwise, make sure that we convert all multiple questions into single questions
-				$existing_questions = array();
+				// Otherwise, make sure that we convert all multiple questions into single questions.
+				$existing_questions = [];
 
-				// Set array of questions that already exist so we can prevent duplicates from appearing
+				// Set array of questions that already exist so we can prevent duplicates from appearing.
 				foreach ( $questions_array as $question ) {
-					if ( 'question' != $question->post_type ) {
+					if ( 'question' !== $question->post_type ) {
 						continue;
 					}
 					$existing_questions[] = $question->ID;
 				}
 
-				// Include only single questions in the return array
+				// Include only single questions in the return array.
 				$questions_loop  = $questions_array;
-				$questions_array = array();
+				$questions_array = [];
 				foreach ( $questions_loop as $k => $question ) {
 
-					// If this is a single question then include it
-					if ( 'question' == $question->post_type ) {
+					// If this is a single question then include it.
+					if ( 'question' === $question->post_type ) {
 						$questions_array[] = $question;
 					} else {
 
-						// If this is a multiple question then get the specified amount of questions from the specified category
-						$question_cat    = intval( get_post_meta( $question->ID, 'category', true ) );
-						$question_number = intval( get_post_meta( $question->ID, 'number', true ) );
+						// If this is a multiple question then get the specified amount of questions from the specified category.
+						$question_cat    = (int) get_post_meta( $question->ID, 'category', true );
+						$question_number = (int) get_post_meta( $question->ID, 'number', true );
+						$quiz_author     = get_post( $quiz_id )->post_author;
 
-						$qargs         = array(
+						$qargs = [
 							'post_type'        => 'question',
 							'posts_per_page'   => $question_number,
 							'orderby'          => $orderby,
-							'tax_query'        => array(
-								array(
+							'tax_query'        => [
+								[
 									'taxonomy' => 'question-category',
 									'field'    => 'term_id',
 									'terms'    => $question_cat,
-								),
-							),
+								],
+							],
 							'post_status'      => $post_status,
 							'suppress_filters' => 0,
 							'post__not_in'     => $existing_questions,
-						);
+						];
+
+						if ( ! user_can( $quiz_author, 'manage_options' ) ) {
+							$qargs['author'] = $quiz_author;
```

Comment:
Since this is a possibly breaking change for some sites that have moved from admin teacher -> teacher teacher, I wonder about putting this around a new filter (feel free to rename 😆 ). 
```
if ( apply_filters( 'sensei_category_questions_only_quiz_author`, true, $quiz_id ) && ! user_can( $quiz_author, 'manage_options' ) ) {
// ...
```

Reply:
I thought about this too but I figured that users could use the `sensei_lesson_quiz_questions` below to achieve the old behaviour and prefered to avoid a new filter. I added a new one `sensei_filter_category_questions_by_author` for this specific part but I can revert it if you think that the existing one is enough.

Reply:
If we were filtering on the `get_post()` args I'd say just that, but I feel like it is much harder to add removed questions than to not remove them. This looks good! One minor thing above.